### PR TITLE
Development

### PR DESCRIPTION
**Pull request by ahmadamira**

### CODE REVIEW

File: index.css [hunk 1]
```diff
@@ -0,0 +1,80 @@
+body {
+  font-family: Arial, sans-serif;
+  margin: 0;
+  padding: 0;
+  background-color: #f0f0f0;
+}
+
+.container {
+  max-width: 1000px;
```
Comment (omardere) [inline index.css:9]: the unit size is px by default. you shouldn't write it

File: index.css [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  margin: 0 auto;
+  padding: 20px;
```
Comment (omardere) [inline index.css:11]: and here

File: index.css [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+h1 {
+  text-align: center;
+}
+
+.input-group {
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 20px;
```
Comment (omardere) [inline index.css:23]: and here
apply it in each px in the file

File: index.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>TODO LIST</title>
+    <script src="index.js"></script>
+
```
Comment (omardere) [inline index.html]: unneeded space

File: index.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    <link rel="stylesheet" href="index.css" />
+  </head>
+  <body>
+    <div class="container">
```
Comment (omardere) [inline index.html:11]: use semantic HTML instead id. And use more indicating name

File: index.html [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      <h1>TODO LIST</h1>
+
+      <div class="input-group">
+        <input type="text" id="taskInput" placeholder="Add a new task" />
+        <button onclick="addTask()">Add Task</button>
+      </div>
+      <div class="input-group">
+        <input type="text" id="searchInput" placeholder="Search for a task" />
```
Comment (omardere) [inline index.html]: used search input. `type=search`

File: index.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        <td>${task.id}</td>
+        <td>${task.description}</td>
+        <td>${task.userId}</td>
+        <td>${task.status}</td>
+        <td class="actions">
+            <button class="done-button" onclick="markDone(this)">Done</button>
+            <button class="delete-button" onclick="deleteTask(this)">Delete</button>
+        </td>
+    `;
+    taskId = Math.max(taskId, task.id + 1);
```
Comment (omardere) [inline index.js]: can you use the size of the array instead of session Storage

File: index.js [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+  loadTasksFromSessionStorage();
+  updateTotalTodosCount();
+};
+
+async function fetchTasksFromAPI() {
+  try {
+    const response = await fetch("https://dummyjson.com/todos");
+    const tasks = await response.json();
+
+    console.log(tasks);
```
Comment (omardere) [inline index.js]: pls remove this

File: index.js [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+  for (let i = 0; i < rows.length; i++) {
+    const taskDescription = rows[i].cells[1].textContent.toLowerCase();
+
+    if (taskDescription.includes(searchText)) {
+      rows[i].style.display = "";
+    } else {
+      rows[i].style.display = "none";
+    }
+  }
+}
```
Comment (omardere) [inline index.js]: can you use the array utils instead of that

**Review by omardere — COMMENTED** [review]
I see you use same logic multi time. like when you add tasks.
you can separate shared logic in new file called utils and used it in your code

File: index.css (hunk 1 above)
Comment (ahmadamira) [reply index.css:9]: it dose not work when i remove the px

**Pull request merged**